### ISSUE TITLE
Add support for GCP Cloud Storage

### ISSUE THREAD

**Issue by alikhajeh1** [infracost/infracost#231]
Pricing details: https://cloud.google.com/storage/pricing
Terraform resources: google_storage*

This can be a complicated resource to add so it might help to study how aws_s3_* ones were done in Infracost, e.g. https://github.com/infracost/infracost/blob/master/internal/providers/terraform/aws/s3_bucket.go and join [our Discord](https://discord.gg/rXCTaH3) if you want to work on this.

**Comment by sinabakh** [comment]
I'll give it a go.

**Comment by alikhajeh1** [comment]
Thanks @sinabakh! May the force be with you, it's a big one :) ping the community Slack chat or comment here if you need help

**Comment by sinabakh** [comment]
I think I've found some kind of logical error in Terraform provider for this resource. So I just wanted to ask your opinion regarding this.

Buckets have a location that indicates where they are located. You can find the list of locations and their region in https://cloud.google.com/storage/docs/locations

They have 3 kinds of locations:
* Regional (like `US-EAST4`)
* Dual-regional (like `US`)
* Multi-regional (like `NAM4`)

On the other hand, storages have 4 main storage class listed in https://cloud.google.com/storage/pricing
They are:
* Standard
* Nearline
* Coldline
* Archive

The problem with the Terraform provider is that in its `storage_class` field description it is mentioned that:
```
(Optional, Default: 'STANDARD') The Storage Class of the new bucket. Supported values include: STANDARD, MULTI_REGIONAL, REGIONAL, NEARLINE, COLDLINE, ARCHIVE
```

As far as I understand, the `MULTI_REGIONAL` and `REGIONAL` can have no meaning cause the location field itself dictates the regionality of the bucket. Like one should not be able to set the location to `US` and set the storage class to `Regional`.

It would be great if someone can take another look at the pricing and the docs to see if I'm missing sth.

**Comment by aliscott** [comment]
@sinabakh okay this took a while to work out. `REGIONAL` and `MULTI_REGIONAL` are legacy. You can't even create them any more in the GCP UI. They map to same pricing as `STANDARD` for that location (see https://cloud.google.com/storage/docs/storage-classes#legacy).

Also this tooltip shows in the GCP UI:
> Standard has replaced Multi-Regional and Regional as the hot storage class in this form. This won't change your bucket's cost or behaviour, but if you rely on the Multi-Regional and Regional storage class names, you can revert to an earlier version of this form.

So:
* `location: US, storage_class: MULTI_REGIONAL == location: US, storage_class: STANDARD`
* `location: US, storage_class: REGIONAL == invalid`
* `location: US-EAST4, storage_class: REGIONAL == location: US-EAST4, storage_class: STANDARD`
* `location: US-EAST4, storage_class: MULTI_REGIONAL == invalid`

**Comment by alikhajeh1** [comment]
Shipped in v0.7.16
